roles/lib_openshift/src/class/oc_adm_ca_server_cert.py

```diff
@@ -0,0 +1,134 @@
+            ext = time.strftime("%Y-%m-%d@%H:%M:%S", time.localtime(time.time()))
+            date_str = "%s_" + "%s" % ext
+
+            if os.path.exists(self.config.config_options['key']['value']):
+                shutil.copy(self.config.config_options['key']['value'],
+                            date_str % self.config.config_options['key']['value'])
+            if os.path.exists(self.config.config_options['cert']['value']):
+                shutil.copy(self.config.config_options['cert']['value'],
+                            date_str % self.config.config_options['cert']['value'])
+
+        options = self.config.to_option_list()
+
+        cmd = ['ca', 'create-server-cert']
+        cmd.extend(options)
+
+        return self.openshift_cmd(cmd, oadm=True)
+
+    def exists(self):
+        ''' check whether the certificate exists and has the clusterIP '''
+
+        cert_path = self.config.config_options['cert']['value']
+        if not os.path.exists(cert_path):
+            return False
+
+        # Would prefer pyopenssl but is not installed.
+        # When we verify it is, switch this code
+        # Here is the code to get the subject and the SAN
+        # openssl x509 -text -noout -certopt \
+        #  no_header,no_version,no_serial,no_signame,no_validity,no_issuer,no_pubkey,no_sigdump,no_aux \
+        #  -in /etc/origin/master/registry.crt
+        # Instead of this solution we will use a regex.
+        cert_names = []
+        hostnames = self.config.config_options['hostnames']['value'].split(',')
+        proc = subprocess.Popen(['openssl', 'x509', '-noout', '-text', '-in', cert_path],
+                                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        x509output, _ = proc.communicate()
+        if proc.returncode == 0:
+            regex = re.compile(r"^\s*X509v3 Subject Alternative Name:\s*?\n\s*(.*)\s*\n", re.MULTILINE)
+            match = regex.search(x509output)  # E501
+            for entry in re.split(r", *", match.group(1)):
+                if entry.startswith('DNS') or entry.startswith('IP Address'):
+                    cert_names.append(entry.split(':')[1])
+            # now that we have cert names let's compare
+            cert_set = set(cert_names)
+            hname_set = set(hostnames)
+            if cert_set.issubset(hname_set) and hname_set.issubset(cert_set):
+                return True
+
+        return False
+
+    @staticmethod
+    def run_ansible(params, check_mode):
+        '''run the idempotent ansible code'''
+
+        config = CAServerCertConfig(params['kubeconfig'],
+                                    params['debug'],
+                                    {'cert':          {'value': params['cert'], 'include': True},
+                                     'hostnames':     {'value': ','.join(params['hostnames']), 'include': True},
+                                     'overwrite':     {'value': True, 'include': True},
```

Comment:
Just to be safe, should the value of `overwrite` be set to match the value of `force`?  It might make sense to make `overwrite` an alias for `force` as well.